### ISSUE TITLE
Minecraft Joules machines accept excess power

### ISSUE THREAD

**Issue by roboderpy** [BuildCraft/BuildCraft#4332]
![pic](https://i.dbotthepony.ru/2019/01/javaw_2019-01-24_06-57-35.png)
![pic](https://i.dbotthepony.ru/2019/01/javaw_2019-01-24_07-07-19.png)
![pic](https://i.dbotthepony.ru/2019/01/javaw_2019-01-24_07-08-19.png)
![pic](https://i.dbotthepony.ru/2019/01/javaw_2019-01-24_07-09-56.png)
BuildCraft version: 7.99.21
Forge version: 
Singleplayer or multiplayer: singleplayer
Steps to reproduce: call `receivePower` on buildcraft quarry with value which would overflow internal battery of the quarry. You expect it to return excess value, but you will get `0L` and machine would accept value which overflow it's buffer.

On my side i fixed this as follows:
![pic](https://i.dbotthepony.ru/2019/01/Code_-_Insiders_2019-01-24_07-09-10.png)
![pic](https://i.dbotthepony.ru/2019/01/Code_-_Insiders_2019-01-24_07-09-17.png)

**Comment by AlexIIL** [comment]
Yes, MJ is broken at the moment, but it will all be changed on the future with the MJ update.

(So I won't bother fixing this now)

**Comment by AlexIIL** [comment]
...although I should probably keep this issue open until I actually release the MJ update.

**Comment by roboderpy** [comment]
API is good but implementation is a bit sad. That's the only thing you need to update.

**Comment by AlexIIL** [comment]
Oh er the MJ update will change the way everything works thematically too, so the whole API is being shelved.
